cpp/src/community/detail/refine.cuh

```diff
@@ -0,0 +1,763 @@
+                    });
+
+  // Update cluster weight, weighted degree and cut for edge sources
+
+  edge_src_property_t<GraphViewType, weight_t> src_louvain_cluster_weight_cache(handle);
+  edge_src_property_t<GraphViewType, thrust::tuple<weight_t, weight_t>>
+    src_wdeg_and_cut_to_Louvain_cache(handle);
+
+  if (GraphViewType::is_multi_gpu) {
+    src_louvain_cluster_weight_cache =
+      edge_src_property_t<GraphViewType, weight_t>(handle, graph_view);
+    update_edge_src_property(
+      handle, graph_view, vertex_cluster_weights_v.begin(), src_louvain_cluster_weight_cache);
+
+    src_wdeg_and_cut_to_Louvain_cache =
+      edge_src_property_t<GraphViewType, thrust::tuple<weight_t, weight_t>>(handle, graph_view);
+    update_edge_src_property(
+      handle,
+      graph_view,
+      thrust::make_zip_iterator(thrust::make_tuple(weighted_degree_of_vertices.begin(),
+                                                   weighted_cut_of_vertices_to_louvain.begin())),
+      src_wdeg_and_cut_to_Louvain_cache);
+
+    vertex_cluster_weights_v.resize(0, handle.get_stream());
+    vertex_cluster_weights_v.shrink_to_fit(handle.get_stream());
+
+    weighted_degree_of_vertices.resize(0, handle.get_stream());
+    weighted_degree_of_vertices.shrink_to_fit(handle.get_stream());
+
+    weighted_cut_of_vertices_to_louvain.resize(0, handle.get_stream());
+    weighted_cut_of_vertices_to_louvain.shrink_to_fit(handle.get_stream());
+  }
+
+  //
+  // Assign Lieden community Id for vertices.
+  // Each vertex starts as a singleton community in the leiden partition
+  //
+
+  rmm::device_uvector<vertex_t> leiden_assignment = rmm::device_uvector<vertex_t>(
+    graph_view.local_vertex_partition_range_size(), handle.get_stream());
+
+  detail::sequence_fill(handle.get_stream(),
+                        leiden_assignment.begin(),
+                        leiden_assignment.size(),
+                        graph_view.local_vertex_partition_range_first());
+
+  edge_src_property_t<GraphViewType, vertex_t> src_leiden_assignment_cache(handle);
+  edge_dst_property_t<GraphViewType, vertex_t> dst_leiden_assignment_cache(handle);
+  edge_src_property_t<GraphViewType, uint8_t> src_active_flag_cache(handle);
+
+  kv_store_t<vertex_t, vertex_t, false> leiden_to_louvain_map(leiden_assignment.begin(),
+                                                              leiden_assignment.end(),
+                                                              louvain_assignment.begin(),
+                                                              invalid_vertex_id<vertex_t>::value,
+                                                              invalid_vertex_id<vertex_t>::value,
+                                                              handle.get_stream());
+
+  auto vertex_begin =
+    thrust::make_counting_iterator(graph_view.local_vertex_partition_range_first());
+  auto vertex_end = thrust::make_counting_iterator(graph_view.local_vertex_partition_range_last());
```

Comment:
It is generally more advisable to minimize the scope of variable and avoid generating unnecessary temporaries unless there is a justification for doing this. This variable is declared here and used in Line 502, any reason to do this?

Reply:
Nope. Will be fixed in next commit